### PR TITLE
TST: Add tests for ComplexWarning in astype

### PR DESCRIPTION
**Pull request by kianasun**

### CODE REVIEW

**Comment by mattip** [comment]
What is your motivation for the additional test?

If you are already touching this test, it needs a refactor into separate tests, perhaps using `pytest.mark.paramtrize`

**Comment by kianasun** [comment]
Hi @mattip, I am adding the tests because the documentation says the function will raise `ComplexWarning` when casting from complex to float or int, but I did not find relevant tests. If a refactor is necessary, I can try to use it.

**Comment by kianasun** [comment]
Hi @mattip, I updated the test with `pytest.mark.paramtrize`.

File: numpy/core/tests/test_api.py [hunk 1]
```diff
@@ -289,6 +290,14 @@ class MyNDArray(np.ndarray):
     a = np.array(1000, dtype='i4')
     assert_raises(TypeError, a.astype, 'U1', casting='safe')
 
+@pytest.mark.parametrize("t",
+    np.sctypes['uint'] + np.sctypes['int'] + np.sctypes['float']
+)
+def test_array_astype_warning(t):
+    # test ComplexWarning when casting from complex to float or int
+    a = np.array(10, dtype=np.complex)
+    assert_raises(np.ComplexWarning, a.astype, t)
```
Comment (seberg) [inline numpy/core/tests/test_api.py]: Before we forget it. This should be `assert_warns` we only raise warnings during testing the development version, not in release mode (although I am not quite sure if pytest nowadays can be set up to handle that part). You may need a warnings context.

Comment (kianasun) [reply numpy/core/tests/test_api.py]: Thanks for your reply. I have changed `assert_raises` to `assert_warns`.

**Comment by mattip** [comment]
LGTM.

Strange that I don't see results for checking coverage here, I was wondering if the code path is tested at all.

**Comment by seberg** [comment]
Yeah, an additional test can't really hurt. IIRC I grepped and it looked a bit like there were regression tests about it, but since those are a bit of a random collection, it is nicer here. Anyway, no need to think about it too much, lets put it in. Thanks @kianasun.

**Pull request merged**